python/pysmurf/client/command/smurf_command.py

```diff
@@ -2729,6 +2729,190 @@ def get_jesd_tx_status_valid_cnt(self, bay, num, **kwargs):
+                    f' {max_timeout_sec} seconds.', self.LOG_ERROR)
+                return status
+
+            # Measure how long the process take
+            end_time = time.time()
+
+            self.log(
+                'JESD health check finished after'
+                f' {int(end_time - start_time)} seconds.'
+                f' The final status was {status}.',
+                self.LOG_USER)
+
+            return status
+        else:
+            self.log(
+                'The `SmurfApplication:CheckJesd` and '
+                ' `SmurfApplication:JesdStatus` registers are not'
+                ' implemented for pysmurf core code versions <4.1.0'
+                f' (current version is {pysmurf_version}).',
+                self.LOG_ERROR)
+            return None
+
+    _jesd_status_reg = "JesdStatus"
+
+    def get_jesd_status(self, as_string=True, **kwargs):
+        r"""Gets the status of the Rogue `AppTop.JesdHealth` method.
+
+        Returns the status of a call to the `AppTop.JesdHealth`
+        method.  States are:
+
+        * "Unlocked" : The `AppTop.JesdHealth` command has been run
+          and has reported that the JESD were unlocked (**that's
+          bad!**).
+        * "Locked" : The `AppTop.JesdHealth` command has been run and
+          has reported that the JESD were locked (that's good!).
+        * "Checking" : The `AppTop.JesdHealth` command is in progress.
+        * "Not found" : The Rogue ZIP file currently in use does not
+          contain the `AppTop.JesdHealth` method.
+        * `None` : The `SmurfApplication:JesdStatus` register is not
+          implemented in pysmurf core code versions <4.1.0.
+
+        Args
+        ----
+        as_string : bool, optional, default True
+            Returns the data as a string.
+        \**kwargs
+            Arbitrary keyword arguments.  Passed directly to
+            all `_caget` calls.
+
+        Returns
+        -------
+        status : str or None
+            The status of the Rogue `AppTop.JesdHealth` method.
+            Returns None for pysmurf core code versions <4.1.0 where
+            the `SmurfApplication:JesdStatus` register is not
+            implemented.
+
+        See Also
+        --------
+        :func:`set_check_jesd` : Gets the status of the Rogue
+              `AppTop.JesdHealth` method.
+
+        """
+        # strip any commit info off the end of the pysmurf version
+        # string
+        pysmurf_version = self.get_pysmurf_version(**kwargs).split('+')[0]
+
+        # Extra registers were added to allow confirmation of JESD
+        # lock for pysmurf versions >=4.1.0.
+        # https://github.com/slaclab/pysmurf/issues/467 for more
+        # details.
+        if version.parse(pysmurf_version) >= version.parse('4.1.0'):
+            status =  self._caget(
+                self.smurf_application + self._jesd_status_reg,
+                as_string=as_string, **kwargs)
+
+            if status == 'Not found':
```

Comment:
If you call this method with `S.get_jesd_status(as_string=False)` this statement will failed, as status will be a number.